openedx/adg/lms/applications/management/commands/tests/test_update_is_prerequisite_courses_passed.py

```diff
@@ -0,0 +1,301 @@
+"""
+Tests to check if `update_is_prerequisite_courses_passed` command actually updates the `is_prerequisite_courses_passed`
+flag in Application Hub for each eligible user
+"""
+from datetime import timedelta
+
+import pytest
+from django.core.management import call_command
+from django.utils import timezone
+
+from openedx.adg.common.course_meta.tests.factories import CourseMetaFactory
+from openedx.adg.lms.applications.management.commands import update_is_prerequisite_courses_passed as command_module
+from openedx.adg.lms.applications.models import ApplicationHub
+from openedx.core.djangoapps.content.course_overviews.tests.factories import CourseOverviewFactory
+from student.tests.factories import CourseEnrollmentFactory, UserFactory
+
+
+@pytest.fixture
+def create_users(request):
+    """
+    This fixture creates users in batch based on the size sent in parameters
+    """
+    return UserFactory.create_batch(size=request.param)
+
+
+@pytest.fixture
+def create_courses(request):
+    """
+    This fixture creates courses equal to the size of the boolean list and makes them prerequisite depending on the
+    boolean value at each index
+    """
+    courses = list()
+    prereq_courses = list()
+
+    for is_prereq in request.param:
+        course = CourseOverviewFactory(
+            end_date=timezone.now() + timedelta(days=1),
+            start_date=timezone.now(),
+        )
+
+        if is_prereq:
+            CourseMetaFactory(course=course, is_prereq=is_prereq)
+            prereq_courses.append(course)
+
+        courses.append(course)
+
+    return courses, prereq_courses
+
+
+@pytest.fixture
+def enroll_users(request, create_users, create_courses):  # pylint: disable=redefined-outer-name
+    """
+    This fixture enrolls a user into courses sent by the `create_users` and `create_courses` fixtures respectively and
+    makes the enrollments active/inactive depending on the boolean value at each index of the list sent as a parameter
+    """
+    activations = request.param
+    courses, prerequisites = create_courses
+    user = create_users.pop()
+    enrollments = list()
+
+    for i, is_active in enumerate(activations):
+        enrollment = CourseEnrollmentFactory(
+            course_id=courses[i].id,
+            user=user,
+            is_active=is_active
+        )
+        enrollments.append(enrollment)
+
+    return prerequisites, user
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('create_users', [1], indirect=True)
+@pytest.mark.parametrize('create_courses', [[True]], indirect=True)
+@pytest.mark.parametrize('enroll_users, expected_result', [([False], 0), ([True], 1)], indirect=['enroll_users'],
+                         ids=['enrolled', 'not_enrolled'])
+def test_get_minimal_users_to_be_checked_for_update_for_active_and_inactive_enrollments(enroll_users, expected_result):
+    # pylint: disable=redefined-outer-name
```

Comment:
Instead of disabling these errors, solve them by giving a `name` to your fixtures.
```
@pytest.fixture(name='test_users')
def enroll_users(request, create_users, create_courses):
```

And then pass whatever name you've given it, to the tests where that particular fixture is needed.